File: src/models/data-sheet.ts

export class Topic {
    constructor(
        public topic: string,
        public order: number,
        public level: number,
        public title: string,
        public details: string,
        public notYetAvailable: boolean,
        public allowISO: string,
        public denyISO: string
    ) {}
}

export class DataSheetLineOption {
    constructor(
        public edition: string
    ) {}

    public availability: string = "";
}

export class DataSheetLine {
    constructor(
        line: DataSheetLine
    ) { Object.assign(this, line);}

    public topic: string;
    public order: number;
    public level: number;
    public title: string;
    public details: string;
    public notYetAvailable: boolean;
    public allowISO: string;
    public denyISO: string;
    public options: DataSheetLineOption[] = [];

    public included: boolean;
}

export class DataSheet {
    constructor(
        public name: string
    ) {}
    public lines: DataSheetLine[] = [];
}

export class CSVDataSheetLine {
    constructor(
        line: CSVDataSheetLine
    ) { Object.assign(this, line);}

    public level: number;
    public title: string;
    public details: string;
}

export class CSVDataSheet {
    constructor(
        public name: string
    ) {}
    public lines: CSVDataSheetLine[] = [];
}

File: Models/DataSheet.cs
using System.Collections.Generic;

namespace Configurator.Models
 {
    public class DataSheetLineOption
    {
        public string edition {get;set;}
        public string availability {get;set;}
    }

    public class DataSheetLine
    {
        public string Topic { get; set; }
        public string Order { get; set; }
        public int Level { get; set; }
        public string Title { get; set; }
        public string Details { get; set; }
        public bool NotYetAvailable { get; set; }
        public string allowISO {get;set;}
        public string denyISO {get;set;}
        public List<DataSheetLineOption> options {get;set;} = new List<DataSheetLineOption>();
    }

    public class DataSheet
    {
        public string Name { get; set; }
        public DataSheetLine[] Lines { get; set; }
    }
 }